clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -476,15 +476,38 @@ static void addImpliedArgs(const llvm::Triple &Triple,
                            llvm::opt::ArgStringList &CmdArgs) {
   // Current implied args are for debug information and disabling of
   // optimizations.
-  // TODO: Add support for other architectures (gen, x86_64) as those are
-  // being defined.
-  if (Triple.getSubArch() == llvm::Triple::NoSubArch ||
-      Triple.getSubArch() == llvm::Triple::SPIRSubArch_fpga) {
+  auto addArgs = [&](llvm::opt::ArgStringList &CurArgList, bool IsGen = false) {
```

Comment:
This part of the code is quite generous with `*Args` naming. Personally, I'd rather see a more expressive name, something like `addOptDebugArgs()`.

Reply:
Done.